### PR TITLE
Support Author-namedspaced Thumbnails and Revamp building_map_generator

### PR DESCRIPTION
**Pull request by methylDragon**
## Unify `building_map_generator`
- This PR unifies all `building_map_generators` into a single one and implements `argparse` command line argument handling.
- Additionally, the `gazebo` generator will call `pit_crew`  to obtain missing models automatically when generating .world files.

## Support Author-namespaced Thumbnails
- Additionally, the thumbnail generators have been revamped to support the new author-namedspaced thumbnail directory structure as shown in https://github.com/osrf/traffic_editor_assets/pull/2
- Lastly, `output_dir` for `crop.py` has been set by default to the `~/.traffic_editor/assets/thumbnails/images/cropped/` directory, for parity with the automatic clone on build of `traffic_editor` of `traffic_editor_assets`

## Notes
- A new package dependency for `traffic_editor` has been added. Specifically for `building_map_tools`, in order to ensure that `traffic_editor` has access to the `pit_crew` dependency.

- **Request for test**: I can't test the `traffic_editor` thumbnail generation pipeline due to having a different version of OpenCV (4.0.0), could someone test it out?

### CODE REVIEW

File: traffic_editor/thumbnail_generator/README.md [hunk 1]
```diff
@@ -49,15 +49,15 @@ The orthogonal views of the models with a green screen will have been saved in t
...
+./scripts/top_view_generator.py test/model_list.yaml images/white
 ```
 
 Similarly, the folder `images/white/` will have been populated when the script finishes.
 
 The contours of the models are then extracted from the green-screened images using OpenCV to generate a transparency mask, used for cropping the white-background images, so that it stays centered and is as small as possible. This can be done by calling,
 
 ```bash
-./scripts/crop.py -m test/model_list.yaml -g images/green -w images/white -o images/cropped
+./scripts/crop.py test/model_list.yaml images/cropped -g images/green -w images/white
```
Comment (aaronchongth) [inline traffic_editor/thumbnail_generator/README.md:60]: the `-o` argument is still required in the source code, were you intending to remove it?

**Comment by methylDragon** [comment]
Oh, whoops!

Output dir should be optional, I'll update the readme accordingly.

**Comment by methylDragon** [comment]
Oh, whoops!

Output dir should be optional, I'll update the readme accordingly.

**Comment by aaronchongth** [comment]
here
https://github.com/methylDragon/traffic_editor/blob/ch3/author-namespaced-thumbnails/traffic_editor/thumbnail_generator/scripts/crop.py#L84

**Comment by methylDragon** [comment]
Yep! When you have `-` and `--` in argparse, it specifies an optional argument

**Comment by methylDragon** [comment]
Or do you mean the model file?

**Comment by methylDragon** [comment]
image

**Comment by methylDragon** [comment]
Clarification:
Specifying an output directory is optional, if you want to specify one, `-o` or `--output_dir` should be specified

**Comment by methylDragon** [comment]
:+1:  I can't merge it due to no write access though haha

**Pull request merged**